bindings/ergo-lib-wasm/src/rest/api.rs

```diff
@@ -14,3 +15,72 @@ pub async fn get_info(node: NodeConf) -> Result<JsValue, JsValue> {
         .map_err(to_js)
         .map(|info| JsValue::from_str(&info.name))
 }
+
+#[wasm_bindgen]
+/// GET on /nipopow/proof/{minChainLength}/{suffixLength}/{headerId} endpoint
+pub async fn get_nipopow_proof_by_header_id(
+    node: NodeConf,
+    min_chain_length: u32,
+    suffix_len: u32,
+    header_id: BlockId,
+) -> Result<NipopowProof, JsValue> {
+    ergo_lib::ergo_rest::api::node::get_nipopow_proof_by_header_id(
+        node.into(),
+        min_chain_length,
+        suffix_len,
+        header_id.into(),
+    )
+    .await
+    .map_err(to_js)
+    .map(NipopowProof::from)
+}
+
+/// List of peer urls returned from `peer_discovery`. We need this wrapper struct because the
+/// `wasm_bindgen` macro currently cannot deal with `Result<Box<[T]>, JsValue>`, for any value `T`
+/// that can be converted into a `JsValue` (`Result<Box<[web_sys::Url]>, JsValue>` would be a
+/// convenient return type for `peer_discovery`).
+#[wasm_bindgen]
+pub struct PeerUrls(Vec<web_sys::Url>);
+
+#[wasm_bindgen]
+impl PeerUrls {
+    /// Returns the number of elements in the collection
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Returns the element of the collection with a given index
+    pub fn get(&self, index: usize) -> web_sys::Url {
+        self.0[index].clone()
+    }
+}
+
+/// Stub that can be removed once `bounded-vec` is updated with this type
```

Comment:
Got this stub in various places. Perhaps we can bump the version for `bounded-vec`?

Reply:
Yes, please do.

Reply:
Oh, I think you meant that I should release a new version of bounded-vec, right? :) I'll do it ASAP.